numpy/f2py/tests/test_crackfortran.py

```diff
@@ -33,3 +36,50 @@ def test_module(self):
         self.module.subc([w, k])
         assert_array_equal(k, w + 1)
         assert self.module.t0(23) == b'2'
+
+class TestPublicPrivate():
+    def test_defaultPrivate(self):
+        with tempfile.NamedTemporaryFile() as ff:
+            ff.write(textwrap.dedent("""\
```

Comment:
Can drop the flush by using:
```suggestion
    def test_defaultPrivate(self, tmp_path):
        f_path = tmp_path / "mod.f"
        with f_path.open("w") as ff:
            ff.write(textwrap.dedent("""\
```

Reply:
In this construction, will the file be deleted on test completion, or on test failure ? That's why I used a NamedTemporaryFile(), so like that I don't have to bother on the life cycle of the generated file.

Reply:
Probably better as `tmpdir`, see the [pytest documentation](https://docs.pytest.org/en/latest/tmpdir.html). The past paragraph talks about the removal strategy. I like it better than the `NamedTemporaryFile` one since it leave the files around for a few test runs in case I want to view them.

Reply:
@mattip: Why `tmpdir` over `tmp_path`?

Reply:
Ok, thanks for the explanation, I'm a noob in Python. I'll change for the suggestion this afternoon.

Reply:
Ahh, sorry, we are good with pathlib on all supported versions of python. So `tmp_path` is the way to go.

Reply:
Here it is. I've modified the tests as suggested to use the `tmp_path` fixture.